src/lib/apiTypesV2/MqttInfo.cpp

```diff
@@ -0,0 +1,97 @@
+/*
```

Comment:
Right now, `MqttInfo` does not get compiled, I left it in the commit to start talking about what is the best way to handle the MQTT notification information.

As of now, `HttpInfo` is very commonly used throughout the Orion code. Creating `MqttInfo` which would hold the MQTT information and would be carried around as `HttpInfo`, sounds reasonable but this means we have to touch lots of lines of code.

Another option might be to carry the MQTT information in `HttpInfo` but this would make the code less understandable and maintainable. 

Creating something like `ConnectionInfo` might be nice but then again we have to touch maybe even more lines of code than the first option.

What do you think?

Reply:
I think using `MqttInfo` is a good idea. Could you elaborate on the impact of such solution? I mean, why we have to touch lots of line of code in this case? As far as I understand, the main points to touch would be:

* Subscription parsing logic, to properly generated `MqttInfo` object based on JSON payload.
* Subscripton rendering logic, lo properly render JSON corresponding to `MqttInfo` logic.
* Notification logic, where the existence of `MqttInfo` instead of `HttpInfo` would divert to the part of the logic which actually deal with publication in the MQTT broker.

The logic sholud be quite parallel to the one already existing for `HttpInfo`. I mean, look for places in the code where `HttpInfo` is used probably would point out the points in which `MqttInfo` has to be introduced.

However, I could be missing something and be underestimating the difficulty... What do you think?

Reply:
> look for places in the code where HttpInfo is used

Yes, that's was what I did to estimate the impact. Please allow me to detail it further. Among `.h` and `.cpp` files, `HttpInfo` is referenced in 429 unique lines of code among 33 files.

This may not seem much but for example, although having only **8 unique lines of reference** to `HttpInfo`, the part of the code that is direcly related to parsing of `HttpInfo` in `parseSubscription.cpp` is **more than 150 lines**.

Consequently, I believe it is safe to assume that the impact would not be small in terms of number of lines and files we need to touch.

Nevertheless, I don't see any clean way to achieve this with a smaller impact. So I say, let's go with MqttInfo.

Reply:
> Nevertheless, I don't see any clean way to achieve this with a smaller impact. So I say, let's go with MqttInfo.

Yes, I agree. Thus, let's go with `HttpInfo` (I understand that in the next PR).

Reply:
Yes I plan it for the next PR, otherwise this one will be a bit hard to manage :)

Reply:
NTC? Or is there anything pending to decide?

Reply:
NTC for this PR.